src/Extension/FluentVersionedExtension.php

```diff
@@ -595,4 +634,331 @@ public function updateLocalisationTabConfig(GridFieldConfig $config)
             PublishAction::create(),
         ]);
     }
+
+    /**
+     * Localise archived state
+     * Extension point in @see Versioned::isArchived()
+     *
+     * @param bool $isArchived
+     */
+    public function updateIsArchived(bool &$isArchived): void
+    {
+        $locale = FluentState::singleton()->getLocale();
+
+        if (!$locale) {
+            return;
+        }
+
+        $isArchived = $this->owner->isArchivedInLocale();
+    }
+
+    /**
+     * @param string|null $locale
+     * @return bool
+     */
+    public function isArchivedInLocale(?string $locale = null): bool
+    {
+        $locale = $locale ?: FluentState::singleton()->getLocale();
+
+        if (!$locale) {
+            return false;
+        }
+
+        $owner = $this->owner;
+        $class = $owner->ClassName;
+        $id = $owner->ID ?: $owner->OldID;
+
+        if (!$id) {
+            return false;
```

Comment:
Should this also be `null`, rather than `false`?

Reply:
I think this should be `false` as no `ID` indicates a new object which doesn't have an archive.